### DISCUSSION TITLE
[addon-docs] Previous/Next links in docs.

### DISCUSSION THREAD

**Discussion by folz** [storybookjs/storybook#23059] [Ideas]
**Is your feature request related to a problem? Please describe.**

I would like docs stories to link to their Previous and Next stories at either the top or bottom of the page.

I tried searching for variations of "storybook" "addon-docs" "previous next" "link", but wasn't able to find documentation that this feature already exists.

**Describe the solution you'd like**

Here is a screenshot from Docusaurus showing the ideal behavior:

<img width="747" alt="Screen Shot 2020-01-29 at 1 36 06 PM" src="https://user-images.githubusercontent.com/299281/73399659-7eecc900-429c-11ea-9a4a-45edaf6a5a93.png">

For Storybook, given a sidebar of 

- Guide
  - Getting Started
  - Installation
- Components
  - Button
  - Card

I would like the Guide|Installation docs to include

```
[ Previous                ]   [                Next ]
[ Guide - Getting Started ]   [ Components - Button ]
```

at either/both of the top and bottom of the docs.

**Describe alternatives you've considered**

Storybook supports linking to stories, so I could manually include the previous and next buttons for each docs story. However, this is manual and prone to becoming outdated. Ideally we would look at the storybook stories context, but I don't know enough about storybook's internals to know how to do this.

**Are you able to assist bring the feature to reality?**

Yes, if you can point me to the relevant code.

**Comment by shilman** [comment]
Hi @folz this would be an awesome example of a fairly simple docblock to write. I'm on the hook for documenting `addon-docs` as part of the 6.0 milestone, so once there's a dev guide it would be amazing if you could kick the tires on that and see how hard it is to implement this. I'd expect it shouldn't be too hard to extract this information automatically and create the links.

**Comment by folz** [comment]
For sure, that sounds good to me! Is there a place I can follow for updates to 6.0, or would you be able to tag me in this issue once that's released?

**Comment by shilman** [comment]
@folz absolutely. i'll tag you here when it's ready.

in terms of 6.0 here's a general issue (still basically empty): https://github.com/storybookjs/storybook/issues/9311

and for doc blocks documentation there isn't an issue yet. here's the closest thing to track https://github.com/storybookjs/storybook/issues/8379

**Comment by stale** [comment]
Hi everyone! Seems like there hasn't been much going on in this issue lately. If there are still questions, comments, or bugs, please feel free to continue the discussion. Unfortunately, we don't have time to get to every issue. We are always open to contributions so please send us a pull request if you would like to help. Inactive issues will be closed after 30 days. Thanks!

**Comment by shilman** [comment]
Converting this to a discussion. We’re reserving feature issues for things that are on our [short-term roadmap](https://github.com/storybookjs/storybook/milestones). We’ll check back periodically to see whether this is a good candidate to be added. If this feature would be useful to you, please upvote!

**Comment by AngusMacrae** [comment]
This would be an excellent feature to add. We have implemented it in our codebase in a hacky way but a proper official DocBlock for it would be great.

**Comment by kerryj89** [comment]
Until we get a proper block for this basic feature, here's a block you can place in your MDX files.  Might have to tweak it some for your setup or Storybook version (I'm on version 8).  Tried using `items.json` initially and figuring out using API but it was a bit overwhelming... sometimes it's just easier to work off rendered content in the sidebar.

```tsx
export const Footer = ({
  nextLink,
  prevLink,
  nextLinkText,
  prevLinkText
}: {
  nextLink: string;
  prevLink: string;
  nextLinkText: string;
  prevLinkText: string;
}) => {
  var findAdjacentSibling = function (elem, selector, direction: 'prev' | 'next') {
    // Get the next sibling element
    var sibling = elem[direction === 'prev' ? 'previousElementSibling' : 'nextElementSibling'];

    if (!selector) return sibling;

    while (sibling) {
      if (sibling.matches(selector)) return sibling;
      sibling = sibling[direction === 'prev' ? 'previousElementSibling' : 'nextElementSibling'];
    }
  };

  const getAdjacentPageInSidebar = (direction: 'prev' | 'next') => {
    // From parent iframe
    const sidebar = window.parent.document.querySelector('#storybook-explorer-tree');
    if (!sidebar) return null;
    const sidebarActiveItem = sidebar.querySelector('[data-selected="true"]');
    if (!sidebarActiveItem) return null;
    const sidebarAdjacentToActiveItem = findAdjacentSibling(sidebarActiveItem, '.sidebar-item', direction);
    if (!sidebarAdjacentToActiveItem) return null;
    const sidebarAdjacentToActiveItemLink = sidebarAdjacentToActiveItem.querySelector('a');
    if (!sidebarAdjacentToActiveItemLink) return null;
    console.log(sidebarAdjacentToActiveItemLink.getAttribute('href'));
    return sidebarAdjacentToActiveItemLink;
  };

  const prevDocLink = getAdjacentPageInSidebar('prev');
  const nextDocLink = getAdjacentPageInSidebar('next');
  const prevDocLinkText = prevDocLink?.textContent;
  const nextDocLinkText = nextDocLink?.textContent;
  const prevDocLinkHref = prevDocLink?.getAttribute('href');
  const nextDocLinkHref = nextDocLink?.getAttribute('href');

  return (
    <>
      {(prevLink || prevDocLinkHref) && (
        <a href={prevLink ?? prevDocLinkHref}>
          {prevLinkText || prevDocLinkText}
        </a>
      )}
   
      {(nextLink || nextDocLinkHref) && (
        <a href={nextLink ?? nextDocLinkHref}>
          {nextLinkText || nextDocLinkText}
        </a>
      )}
    </>
  );
};

```

Then at the bottom of your MDX you would do `<Footer />`
